litestar/_openapi/request_body.py

```diff
@@ -2,33 +2,65 @@
 
 from typing import TYPE_CHECKING
 
+from litestar._openapi.schema_generation import SchemaCreator
 from litestar.enums import RequestEncodingType
 from litestar.openapi.spec.media_type import OpenAPIMediaType
 from litestar.openapi.spec.request_body import RequestBody
 from litestar.params import BodyKwarg
 
-__all__ = ("create_request_body",)
+__all__ = ("RequestBodyFactory",)
 
 
 if TYPE_CHECKING:
-    from litestar._openapi.schema_generation import SchemaCreator
-    from litestar.handlers import BaseRouteHandler
+    from litestar._openapi.datastructures import OpenAPIContext
+    from litestar.dto import AbstractDTO
     from litestar.typing import FieldDefinition
 
 
-def create_request_body(
-    route_handler: BaseRouteHandler, field_definition: FieldDefinition, schema_creator: SchemaCreator
-) -> RequestBody | None:
-    """Create a RequestBody model for the given RouteHandler or return None."""
-    media_type: RequestEncodingType | str = RequestEncodingType.JSON
-    if isinstance(field_definition.kwarg_definition, BodyKwarg) and field_definition.kwarg_definition.media_type:
-        media_type = field_definition.kwarg_definition.media_type
+class RequestBodyFactory:
```

Comment:
Same as above, but even stronger IMO, since it only *has* one method. I guess consistency is nice here, but if the classes were wrapped in a function call as suggested above, that wouldn't be an issue either :)

Reply:
Yeah, consistency with the other types was my reasoning, will prob remove after adding the functional interface as you suggest.